### PR TITLE
Astroquery/JPL Horizons deprecation  fix

### LINKED ISSUE #1098: Astroquery deprecation warning in tests
A new astroquery deprecation warning popped up in tests:

```
pyuvdata/tests/test_utils.py::test_jphl_lookup
pyuvdata/uvdata/tests/test_uvdata.py::test_phase_dict_helper_jpl_lookup_append
  /Users/bryna/opt/anaconda3/envs/rasg/lib/python3.8/site-packages/astroquery/jplhorizons/core.py:113: AstropyDeprecationWarning: ``id_type``s 'majorbody' and 'id' are deprecated and replaced with ``None``, which has the same functionality.
    warnings.warn("``id_type``s 'majorbody' and 'id' are deprecated "
```

### PR DESCRIPTION
**Pull request by kartographer**
Deals with a pesky deprecation warning from astroquery. 

## Description

- Updated the required version of astroquery to v0.4.4, following an  API change that was made for handling requests to the JPL Horizons service.
- Dropped the use of `id_type="majorbody"` in favor of `id_type=None`, which is preferred when dealing with major SSBs or ID numbers for targets.

## Motivation and Context

The astroquery API for JPL-Horizons has undergone major revision, and with it has come a new deprecation warning about which keywords are allowed for `id_type`. The changes are not entirely backward-compatible, and as there was no prior version requirement for astroquery, the updated requirement of 0.4.4 requires development against only the updated (v0.? -> v1.0) JPL Horizons API moving foward.

Closes #1098

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Bug fix checklist:
- [x] My fix includes a new test that breaks as a result of the bug (if possible).
- [x] All new and existing tests pass.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/CHANGELOG.md).

Build or continuous integration change checklist:
- [x] If required or optional dependencies have changed (including version numbers), I have updated the readme to reflect this.
- [ ] If this is a new CI setup, I have added the associated badge to the readme and to references/make_index.py (if appropriate).

### CODE REVIEW

File: environment.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ dependencies:
   - python>3.6
   - astropy>=4.2.1
   - astropy-healpix
-  - astroquery
+  - astroquery>=0.4.4
```
Comment (bhazelton) [inline environment.yaml:9]: We should add this to the CI yamls as well. And while you're editing this file, can you also update the h5py requirement to `h5py>=3.0`? That change was made in PR #1097 but this file was missed in that PR.

**Review by bhazelton — COMMENTED** [review]
Looks good, I'd just like it updated in the CI yamls as well.

File: environment.yaml (hunk 1 above)
Comment (kartographer) [reply environment.yaml:9]: I've fixed the h5py verisons (in f52ad1a6 and cbc2e4cf). As of right now, there is only one astroquery dependency listed in the CI yamls, which is under `ci/pyuvdata_tests.yaml`. Comically enough, this is _already_ listed as `astroquery>=0.4.4`, so there was nothing to do there. I could add that to the other yamls as well, but none of the other CI builds seem to include the optional dependencies (which astroquery is listed as), so I'd just like to double-check with you to make sure that's what you really want.

Comment (bhazelton) [reply environment.yaml:9]: Ah, I didn't realize it was already set to 0.4.4 in our main CI yaml. I think it should also be added to the `pyuvdata_tests_windows` one but none of the others. The windows one should have all the optional dependencies that are supported on Windows. Thanks!

**Review by bhazelton — APPROVED** [review]
Thank you for all this cleanup!

**Pull request merged**